components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java

```diff
@@ -0,0 +1,174 @@
+        }
+        return true;
+    }
+
+    private void populateLogFileInfo(MessageContext synCtx) {
+
+        List<LogFileInfo> logFileInfoList = Utils.getLogFileInfoList();
+        JSONObject jsonBody = Utils.createJSONList(logFileInfoList.size());
+
+        for (LogFileInfo logFileInfo : logFileInfoList) {
+            JSONObject logfileObject = new JSONObject();
+            logfileObject.put("FileName", logFileInfo.getLogName());
+            logfileObject.put("Date", logFileInfo.getLogDate());
+            logfileObject.put("Size", logFileInfo.getFileSize());
+            jsonBody.getJSONArray(Constants.LIST).put(logfileObject);
+        }
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+    }
+
+    private void populateFileContent(MessageContext synCtx, String pathParameter) {
+
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+
+        DataHandler dataHandler = downloadArchivedLogFiles(pathParameter);
+        if (dataHandler != null) {
+            try {
+                InputStream fileInput = dataHandler.getInputStream();
+                SOAPFactory factory = OMAbstractFactory.getSOAP12Factory();
+                SOAPEnvelope env = factory.getDefaultEnvelope();
+                OMNamespace ns =
+                        factory.createOMNamespace(RelayConstants.BINARY_CONTENT_QNAME.getNamespaceURI(), "ns");
+                OMElement omEle = factory.createOMElement(RelayConstants.BINARY_CONTENT_QNAME.getLocalPart(), ns);
+                StreamingOnRequestDataSource ds = new StreamingOnRequestDataSource(fileInput);
+                dataHandler = new DataHandler(ds);
+                OMText textData = factory.createOMText(dataHandler, true);
+                omEle.addChild(textData);
+                env.getBody().addChild(omEle);
+                synCtx.setEnvelope(env);
+                axis2MessageContext.removeProperty(NO_ENTITY_BODY);
+                axis2MessageContext.setProperty(Constants.MESSAGE_TYPE, "application/octet-stream");
+                axis2MessageContext.setProperty(Constants.CONTENT_TYPE, "application/txt");
+            } catch (AxisFault e) {
+                log.error("Error occurred while creating the response", e);
+                sendFaultResponse(axis2MessageContext);
+            } catch (IOException e) {
+                log.error("Error occurred while reading the input stream", e);
+                sendFaultResponse(axis2MessageContext);
+            }
+        } else {
+            sendFaultResponse(axis2MessageContext);
+        }
+    }
+
+    private void sendFaultResponse(org.apache.axis2.context.MessageContext axis2MessageContext) {
```

Comment:
Can we move this method to a utils class? Other resources would need this as well. WDYT?

Reply:
Other resources already handle this then and there. Here I had to add a new method since it repeated several times.